config/grunt/task-config/shell.js

```diff
@@ -13,10 +13,11 @@ module.exports = function( grunt ) {
 	 * @returns {void}
 	 */
 	function throwUncommittedChangesError( error, stdout, stderr, callback ) {
+		const msg = "You have uncommitted changes. Commit, stash or reset the above files.";
 		if ( stdout ) {
-			throw "You have uncommitted changes. Commit, stash or reset the above files.";
+			throw msg;
 		} else {
-			grunt.log.ok( "You have no uncommitted changes. Continuing..." );
+			grunt.log.error( msg );
 		}
 		callback();
 	}
```

Comment:
This change does not seem to be correct.

If I run `grunt shell:check-for-uncommitted-changes` without any uncommitted changes, I still get the error.

I think this code checks if the `git status --porcelain` command gives any output. Since it only gives output if there are any uncommitted changes, we can check if the output (`stdout`) is empty.

Reply:
Perhaps there are some inconsistencies between Mac/Linux and Windows on how this function is called?